Signal/src/call/PeerConnectionClient.swift

```diff
@@ -246,38 +280,65 @@ class PeerConnectionClient: NSObject, RTCPeerConnectionDelegate, RTCDataChannelD
     public func setLocalVideoEnabled(enabled: Bool) {
         SwiftAssertIsOnMainThread(#function)
 
-        PeerConnectionClient.signalingQueue.async {
-            guard self.peerConnection != nil else {
-                Logger.debug("\(self.TAG) \(#function) Ignoring obsolete event in terminated client")
+        // Define completion early to avoid capturing self within a block
+        // that uses weak self.
+        let completion = { [weak self] in
```

Comment:
Change 3: if we enqueue a closure from an enqueued closure, we risk capturing self when the inner closure is defined.  This effectively captures self in the outer closure, even though we're capturing a weak refernce.  We can avoid this by defining our "completions" (as I'm calling these inner closures) outside of the outer closure.